plugins/processors/starlark/starlark.go

```diff
@@ -93,15 +105,17 @@ func (s *Starlark) Add(metric telegraf.Metric, acc telegraf.Accumulator) error {
 		s.results = s.results[:0]
 	case *common.Metric:
 		m := rv.Unwrap()
-
-		// If the script returned a different metric, mark this metric as
-		// successfully handled.
-		if m != metric {
-			metric.Accept()
+		// If we got the original metric back, use that and drop the new one.
+		// Otherwise mark the original as accepted and use the new metric.
+		if origMetric.HashID() == rv.ID {
+			m.Accept()
+			acc.AddMetric(origMetric)
+		} else {
+			origMetric.Accept()
+			acc.AddMetric(m)
 		}
-		acc.AddMetric(m)
 	case starlark.NoneType:
-		metric.Drop()
+		origMetric.Drop()
```

Comment:
Seems like we drop the metric if it is filtered out, so we should do the same if the original metric is not in the set anymore...

Reply:
I've made this change.